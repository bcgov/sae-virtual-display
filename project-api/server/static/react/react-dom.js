Cannot find "/dist/react-dom.js" in react@0.14.3