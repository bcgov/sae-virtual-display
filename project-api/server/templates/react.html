<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link rel="stylesheet" type="text/css" href="/static/bootstrap/bootstrap.min.css"/>
        <script src="/static/bootstrap/jquery-3.4.1.min.js"></script>
        <script src="/static/bootstrap/bootstrap.min.js"></script>
        <title>DIP Self-Service</title>
    </head>
    <body>
            <div class="container">
                    <h1>Groups</h1>
                    <p>{{ groups }}</p>
                    <div id="root"></div>
                    <script src="/static/react/react.min.js"></script>
                    <script src="/static/react/react-dom.min.js"></script>
                    <script src="/static/react/browser.min.js"></script>
            
                    <script type="text/babel" src="/static/app/greeting.jsx"></script>
            </div>
    </body>
</html>
