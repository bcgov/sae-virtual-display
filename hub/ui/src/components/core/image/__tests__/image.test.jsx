import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import CoreImage from '../';

describe('core/image', () => {
  afterEach(cleanup);

  it('should render', () => {
    const { container } = render(<CoreImage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should handle fluid prop', () => {
    const { container } = render(<CoreImage fluid />);
    expect(container.firstChild).toHaveStyleRule('max-width', '100%');
  });

  it('should always have an alt tag', () => {
    const altText = 'Alternative Text';
    const noAlt = render(<CoreImage />);
    const alt = render(<CoreImage alt={altText} />);

    expect(noAlt.container.firstChild).toHaveAttribute('alt');
    expect(alt.container.firstChild).toHaveAttribute('alt');
  });
});
