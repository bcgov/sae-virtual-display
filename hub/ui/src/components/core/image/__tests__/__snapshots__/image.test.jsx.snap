// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core/image should render 1`] = `
.c0 {
  height: auto;
}

<img
  alt=""
  class="c0"
/>
`;
